src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/InvokeAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,234 @@
+                new ProgressRecord(
+                    0,
+                    Properties.Resources.VaultUpgradeInProgress,
+                    Properties.Resources.WaitingForCompletion);
+
+            do
+            {
+                TestMockSupport.Delay(PSRecoveryServicesClient.TimeToSleepBeforeFetchingJobDetailsAgain);
+                response = this.RecoveryServicesClient.TrackVaultUpgrade(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType);
+
+                elapsedSeconds = DateTime.Now.Subtract(startTime).TotalSeconds;
+                record.PercentComplete = (int)elapsedSeconds * 100 / (int)taskTimeoutInSeconds;
+                this.WriteProgress(record);
+            }
+            while (response.OperationStatus == Constants.InProgress &&
+                elapsedSeconds < taskTimeoutInSeconds &&
+                !this.StopProcessingFlag);
+
+            if (response.OperationStatus == Constants.Completed)
+            {
+                this.WriteResponse(
+                    string.Format(
+                        Properties.Resources.TrackVaultUpgradeJobResult,
+                        response.OperationResult,
+                        response.OperationStatus));
+
+                if (response.OperationResult == Constants.Succeeded)
+                {
+                    this.WriteResponse(
+                        string.Format(
+                            Properties.Resources.VaultUpgradeSucceded,
+                            this.ResourceType,
+                            this.VaultName));
+                }
+                else
+                {
+                    this.WriteResponse(Properties.Resources.VaultUpgradeNotSucceded);
+                }
+            }
+            else if (this.StopProcessingFlag)
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTerminated);
+            }
+            else
+            {
+                this.WriteResponse(Properties.Resources.VaultUpgradeTimedOut);
+            }
+        }
+
+        /// <summary>
+        /// Validating vault upgrade prerequisites.
+        /// </summary>
+        /// <returns>CheckPrerequisites response.</returns> 
+        private CheckVaultUpgradePrerequisitesResponse ValidateVaultUpgradePrerequisites()
+        {
+            CheckVaultUpgradePrerequisitesResponse response =
+                CheckVaultUpgradePrerequisitesResponse.Succeeded;
+
+            this.WriteResponse(Properties.Resources.StartingPrerequisitesCheck);
+
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteObject(Properties.Resources.CheckPrereqSucceeded);
```

Comment:
You should only ever write one object to the output stream in the execution of ProcessRecord